WebAPI/LearningHub.Nhs.Database/Stored Procedures/Resources/GetMyRecentCompletedDashboardResources.sql

```diff
@@ -0,0 +1,101 @@
+					OR (r.ResourceTypeId = 6 AND (sa.CmiCoreLesson_status IN(3,5) OR (ra.ActivityStatusId IN(3, 5))))
+					OR (r.ResourceTypeId = 11 AND ara.Score >= arv.PassMark OR ra.ActivityStatusId IN( 3, 5))
+					OR (r.ResourceTypeId IN (1, 5, 8, 9, 10, 12) AND ra.ActivityStatusId = 3))		
+				GROUP BY ra.ResourceId
+				ORDER BY ResourceActivityId DESC
+
+	SELECT r.Id AS ResourceId
+	,(	SELECT TOP 1 rr.OriginalResourceReferenceId
+		FROM [resources].[ResourceReference] rr
+		JOIN hierarchy.NodePath np on np.id = rr.NodePathId and np.NodeId = n.Id and np.Deleted = 0
+		WHERE rr.ResourceId = rv.ResourceId AND rr.Deleted = 0
+		) AS ResourceReferenceID
+	,r.CurrentResourceVersionId AS ResourceVersionId
+	,r.ResourceTypeId AS ResourceTypeId
+	,rv.Title
+	,rv.Description
+	,CASE 
+		WHEN r.ResourceTypeId = 7	THEN				 
+			(SELECT vrv.DurationInMilliseconds from  [resources].[VideoResourceVersion] vrv WHERE vrv.[ResourceVersionId] = r.CurrentResourceVersionId)
+		WHEN r.ResourceTypeId = 2	THEN				 
+		(SELECT vrv.DurationInMilliseconds from  [resources].[AudioResourceVersion] vrv WHERE vrv.[ResourceVersionId] = r.CurrentResourceVersionId)
+		ELSE 
+		NULL
+		END  AS DurationInMilliseconds
+	,CASE WHEN n.id = 1 THEN NULL ELSE cnv.Name END AS CatalogueName
+	,cnv.Url AS Url
+	,CASE WHEN n.id = 1 THEN NULL ELSE cnv.BadgeUrl END AS BadgeUrl
+	,cnv.RestrictedAccess
+	,CAST(CASE WHEN cnv.RestrictedAccess = 1 AND auth.CatalogueNodeId IS NULL THEN 0 ELSE 1 END AS bit) AS HasAccess
+	,ub.Id AS BookMarkId
+	,CAST(ISNULL(ub.[Deleted], 1) ^ 1 AS BIT) AS IsBookmarked
+	,rvrs.AverageRating
+	,rvrs.RatingCount
+FROM @MyActivity ma 		
+JOIN activity.ResourceActivity ra ON ra.id = ma.ResourceActivityId
+JOIN resources.resourceversion rv ON rv.id = ra.ResourceVersionId AND rv.Deleted = 0		
+JOIN Resources.Resource r ON r.Id = rv.ResourceId
+JOIN hierarchy.Publication p ON rv.PublicationId = p.Id AND p.Deleted = 0
+JOIN resources.ResourceVersionRatingSummary rvrs ON rv.Id = rvrs.ResourceVersionId	AND rvrs.Deleted = 0
+JOIN hierarchy.NodeResource nr ON r.Id = nr.ResourceId AND nr.Deleted = 0
+JOIN hierarchy.Node n ON n.Id = nr.NodeId AND n.Hidden = 0 AND n.Deleted = 0
+JOIN hierarchy.NodePath np ON np.NodeId = n.Id AND np.Deleted = 0 AND np.IsActive = 1
+JOIN hierarchy.NodeVersion nv ON nv.NodeId = np.CatalogueNodeId	AND nv.VersionStatusId = 2 AND nv.Deleted = 0
+JOIN hierarchy.CatalogueNodeVersion cnv ON cnv.NodeVersionId = nv.Id AND cnv.Deleted = 0
+LEFT JOIN hub.UserBookmark ub ON ub.UserId = @UserId AND ub.ResourceReferenceId = (SELECT TOP 1 rr.OriginalResourceReferenceId
+		FROM [resources].[ResourceReference] rr
+		JOIN hierarchy.NodePath np on np.id = rr.NodePathId and np.NodeId = n.Id and np.Deleted = 0
+		WHERE rr.ResourceId = rv.ResourceId AND rr.Deleted = 0)
+LEFT JOIN (  SELECT DISTINCT CatalogueNodeId 
```

Comment:
Is it possible to convert the subqueries into a CTE and utilize that CTE here?

Reply:
I tried this earlier but did not see much difference in terms of performance, I'll refactor to use the CTE then.